metaflow/plugins/aws/batch/batch.py

```diff
@@ -160,6 +168,21 @@ def create_job(
         ).environment_variable(
             "METAFLOW_DEFAULT_METADATA", DEFAULT_METADATA
         )
+        extra_envs = json.loads(os.environ.get(EXTRA_ENVS_VARIABLE_NAME, "[]"))
+        # Only create the cache if there's at least one extra env variable
+        # specifying "tokenProfile"
+        token_cache = (
+            load_cache()
+            if next((True for env in extra_envs if "tokenProfile" in env), False)
```

Comment:
A question of style, but I find something like `any(filter(lambda env: "tokenProfile" in env, extra_envs))` to be clearer

Reply:
Good suggestion